recorder/src/main/java/org/arquillian/cube/q/recorder/TakeNetworkChaosInformation.java

```diff
@@ -0,0 +1,137 @@
+package org.arquillian.cube.q.recorder;
+
+
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import org.arquillian.cube.q.api.Q;
+import org.arquillian.cube.q.toxic.QNetworkChaosToxic;
+import org.arquillian.cube.q.toxic.client.ToxiProxyClient;
+import org.arquillian.cube.q.toxic.event.ToxicCreated;
+
+import org.arquillian.cube.q.toxic.event.ToxicUpdated;
+import org.arquillian.recorder.reporter.ReporterConfiguration;
+import org.arquillian.recorder.reporter.event.PropertyReportEvent;
+import org.arquillian.recorder.reporter.model.entry.FileEntry;
+import org.arquillian.recorder.reporter.model.entry.GroupEntry;
+
+import org.jboss.arquillian.core.api.Event;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.event.suite.After;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class that reports general information about network toxics.
+ */
+public class TakeNetworkChaosInformation {
+
+    private static final String CREATE = "create";
+    private static final String UPDATE = "update";
+
+    @Inject
+    Event<PropertyReportEvent> propertyReportEvent;
+
+    private List<Map<String, Object>> toxics = new ArrayList<>();
+
+    public void captureToxicDetailsAfterCreate(@Observes ToxicCreated toxicCreatedEvent, QNetworkChaosToxic.ToxicAction toxicAction) {
+        final ToxiProxyClient.BaseToxic toxic = toxicCreatedEvent.getToxic();
+        final Q.RunCondition runCondition = toxicCreatedEvent.getRunCondition();
+        addToxicInfotoToxics(toxicAction, toxic, runCondition, CREATE);
+    }
+
+    public void captureToxicDetailsAfterUpdate(@Observes ToxicUpdated toxicUpdatedEvent, QNetworkChaosToxic.ToxicAction toxicAction) {
+        final ToxiProxyClient.BaseToxic toxic = toxicUpdatedEvent.getToxic();
+        addToxicInfotoToxics(toxicAction, toxic, null, UPDATE);
+    }
+
+    public void addToxicInfotoToxics(QNetworkChaosToxic.ToxicAction toxicAction, ToxiProxyClient.BaseToxic toxic, Q.RunCondition runCondition, String phase) {
+        final String actionOn = toxicAction.getName();
+        final String toxicType = toxic.getClass().getSimpleName();
+
+        Map<String, Object> toxicInfo = new LinkedHashMap<>();
+        toxicInfo.put("actionon", actionOn);
```

Comment:
It's already static, but don't want to make it public or package protected to test it. Is it a good idea to make it public to test it?